server/memphis_handlers_tags.go

```diff
@@ -0,0 +1,482 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// This license limiting reselling the software itself "AS IS".
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+package server
+
+import (
+	"context"
+	"errors"
+	"memphis-broker/models"
+	"memphis-broker/utils"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type TagsHandler struct{ S *Server }
+
+const (
+	tagObjectName = "Tag"
+)
+
+func CreateTag(name string, from string, from_name string, background_color string, text_color string) error {
+	name = strings.ToLower(name)
+	exist, _, err := IsTagExist(name)
+	if err != nil {
+		return err
+	}
+	if exist {
+		return errors.New("Tag with that name already exists")
+	}
+	var newTag models.Tag
+	stationArr := []primitive.ObjectID{}
+	schemaArr := []primitive.ObjectID{}
+	userArr := []primitive.ObjectID{}
+	switch from {
+	case "station":
+		station_name, err := StationNameFromStr(from_name)
+		if err != nil {
+			return err
+		}
+		exist, station, err := IsStationExist(station_name)
+		if err != nil {
+			return err
+		}
+		if !exist {
+			return errors.New("Station does not exist")
+		}
+		stationArr = append(stationArr, station.ID)
+		// case "schema":
+		// 	exist, schema, err := IsSchemaExist(from_name)
+		// 	if err != nil {
+		// 		return err
+		// 	}
+		// 	if !exist {
+		// 		return errors.New("Schema with this name does not exist")
+		// 	}
+		// 	schemaArr = append(schemaArr, schema.ID)
+		// case "user":
+		// 	exist, user, err := IsUserExist(from_name)
+		// 	if err != nil {
+		// 		return err
+		// 	}
+		// 	if !exist {
+		// 		return errors.New("User with this name does not exist")
+		// 	}
+		// 	userArr = append(userArr, user.ID)
+	}
+	newTag = models.Tag{
+		ID:       primitive.NewObjectID(),
+		Name:     name,
+		ColorBG:  background_color,
+		ColorTXT: text_color,
+		Stations: stationArr,
+		Schemas:  schemaArr,
+		Users:    userArr,
+	}
+
+	filter := bson.M{"name": newTag.Name, "stations": newTag.Stations, "schemas": newTag.Schemas, "users": newTag.Users}
```

Comment:
remove the arrays of users, schemas and stations from the filter, this will cause the tag to be created every time because mongo don't know to compare arrays